src/libs/OptionsListUtils.js

```diff
@@ -537,12 +537,14 @@ function createOption(accountIDs, personalDetails, report, reportActions = {}, {
  * @returns {Boolean}
  */
 function isSearchStringMatch(searchValue, searchText, participantNames = new Set(), isChatRoom = false) {
-    const searchWords = _.compact(uniqFast([searchValue, ..._.map(searchValue.replace(/,/g, ' ').split(' '), (word) => word.trim())]));
+    const searchWords = new Set(searchValue.replace(/,/g, ' ').split(' '));
     const valueToSearch = searchText && searchText.replace(new RegExp(/&nbsp;/g), '');
-    return _.some(searchWords, (word) => {
+    let matching = true;
+    searchWords.forEach((word) => {
         const matchRegex = new RegExp(Str.escapeForRegExp(word), 'i');
-        return matchRegex.test(valueToSearch) || (!isChatRoom && participantNames.has(word));
+        matching &&= matchRegex.test(valueToSearch) || (!isChatRoom && participantNames.has(word));
```

Comment:
@gedu From what I understood here, once `matching` is set to `false` it won't be set to `true` anymore, right?

Reply:
Yes